pkg/operator/ceph/cluster/controller.go

```diff
@@ -527,8 +527,29 @@ func (c *ClusterController) checkPVPresentInCluster(drivers []string, clusterID
 	return false, nil
 }
 
-// removeFinalizer removes a finalizer
-func removeFinalizer(client client.Client, name types.NamespacedName) error {
+func removeFinalizers(client client.Client, cephCluster *cephv1.CephCluster, name types.NamespacedName) error {
+	// Remove cephcluster finalizer
+	err := removeCephClusterFinalizer(client, name)
```

Comment:
All these three methods seem to be doing the same thing. How about refactoring them to a single method. I'm thinking something like this. Let me know if that makes sense.

```
// remove mon secret finalizer
removeFinalizer(r.client, &corev1.Secret{}, types.NamespacedName{Name: mon.AppName , Namespace: <>})

// call removeFinalizer again for delete other finalizers


func removeFinalizer(client client.Client, obj client.Object, name types.NamespacedName) error {
	err := client.Get(context.TODO(), name, obj)
	if err != nil {
		if kerrors.IsNotFound(err) {
			logger.Debugf("%q resource not found. Ignoring since object must be deleted.", name.Name)
			return nil
		}
		return errors.Wrapf(err, "failed to retrieve resource %q to remove its finalizer", name.Name)
	}

	err = opcontroller.RemoveFinalizer(client, obj)
	if err != nil {
		return errors.Wrapf(err, "failed to remove finalizer from resource %q", name.Name)
	}

	return nil
}
```

Reply:
Yes, this approach will work wisely with a single method :)
Thanks.